source/extensions/filters/http/local_ratelimit/local_ratelimit.cc

```diff
@@ -44,9 +48,21 @@ FilterConfig::FilterConfig(
   if (per_route && !config.has_token_bucket()) {
     throw EnvoyException("local rate limit token bucket must be set for per filter configs");
   }
+
+  // Note: Descriptors work at the route level, as we get rate limit descriptors
+  //       from route's rate limit actions. Hence configuring descriptors at
+  //       global config level does not makes sense.
+  if (!per_route && are_descriptors_configured_) {
+    throw EnvoyException("no descriptors required for global config.");
+  }
+}
```

Comment:
I don't understand this restriction. Global should be fine and can be matched with whatever route is looked up?

Reply:
Just adding why I did this: I thought in global ratelimit config, there won't be any routes to look up...

Reply:
Do we propose to match global descriptors in addition to per route or just ignore the global descriptors? It does make sense to have listener-wide rate limiting.

Reply:
My thinking was just to use per-route if it's there, otherwise use global. Again, I think we can document this explicitly and always add config later. I would keep it simple in the first rev.

Reply:
Yeah, I think that's just the default behavior. Removed this exception.